### PR TITLE
Maven structure

### LINKED ISSUE #569: Change project structure from ant to maven
One priority goal is to change current ant build process and free directory structure to a standard maven build process and directory structure.

### PR DESCRIPTION
**Pull request by Basti890**
Change project structure to Maven build

resolves #569

### CODE REVIEW

**Comment by stweil** [comment]
Hi @Basti890, welcome here. Maybe you want to fix `Ktiodo` in one of your commit messages? Some general hint: there are conventions for commit subjects, like starting them with a verb written in upper case.  See http://chris.beams.io/posts/git-commit/ and many more articles which give hints for writing commit messages.

**Comment by stweil** [comment]
`README.md` needs an update, too, as it contains several references to Ant and still none to Maven.

**Comment by stweil** [comment]
The Maven build starts with warnings:

    [INFO] Scanning for projects...
    [WARNING] 
    [WARNING] Some problems were encountered while building the effective model for org.kitodo.production:kitodo-produktion:war:3.0-SNAPSHOT
    [WARNING] 'version' contains an expression but should be a constant. @ org.kitodo.production:kitodo-produktion:${project.parent.version}, /home/stefan/src/github/stweil/kitodo-production/Kitodo/pom.xml, line 7, column 14
    [WARNING] 
    [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
    [WARNING] 
    [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
    [WARNING]

**Comment by stweil** [comment]
Size with Ant / with Maven in MiB:

    62	dist/kitodo-ce-1.11.2.war
    84	Kitodo/target/kitodo-produktion-3.0-SNAPSHOT.war

I did not examine the reason for this large increase.

**Comment by henning-gerhardt** [comment]
Except words typos I see no reason not to merge this pull request. Other issues could fixed in their own commits / pull requests.

Sebastian is for one week on vacation and he can not react on your mentioned improvements. There are other changes in pipe which go on hold for this time as long as this PR is not merged and two developers can not work on their changes.

File: Kitodo/pom.xml [hunk 1]
```diff
@@ -0,0 +1,291 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<project>
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.kitodo.production</groupId>
+    <artifactId>kitodo-produktion</artifactId>
```
Comment (matthias-ronge) [inline Kitodo/pom.xml]: Please use English naming (“kitodo-production” with “c”, not “k”). When checking out the project with Eclipse, a folder named like this string is generated.

Comment (matthias-ronge) [inline Kitodo/pom.xml]: Also affects the deployment descriptor (URL in Tomcat)

**Comment by matthias-ronge** [comment]
When running the project in Eclipse on Tomcat 7, I get the following error message:

    SLF4J: The requested version 1.5.8 by your slf4j binding is not compatible with [1.5.5, 1.5.6]
    SLF4J: See http://www.slf4j.org/codes.html#version_mismatch for further details.

**Comment by stweil** [comment]
> Except words typos I see no reason not to merge this pull request. Other issues could fixed in their own commits / pull requests.

I understand your requirements, and it's true that most issues can be fixed later. At least the remarks from @matthias-ronge should be addressed immediately. In addition, I'd squash the long commit list into a single commit: it is good to commit often during development, but it is not a good practice to fill the official commit history with all errors made on the way. Especially when bisecting this is really a nuisance.

**Comment by matthias-ronge** [comment]
The additional size of the build seems to be caused by about 60 more JAR files in the WEB-INF/lib directory, with a an additional size of ≈ 25 MB.

**Comment by stweil** [comment]
I already noticed that Maven downloads lots of code when called the first time. Those JAR files are not needed, are they?

File: Kitodo/pom.xml (hunk 1 above)
Comment (henning-gerhardt) [inline Kitodo/pom.xml]: Did I not notice in my development system yet (IDEA with generated WAR file).

**Comment by henning-gerhardt** [comment]
Old lib directory is not cleaned up yet and was not part of this sprint ticket. Clean up is part of next sprint and all your comments.

Squashing commits could be done but on the other side changes should be visible.

So we stop development until Sebastian is back from his vacation and fixed at least the errors mentioned by Matthias.

**Comment by matthias-ronge** [comment]
The error mentioned above seems not to prevent the application from startup—I can successfully launch it.

@henning-gerhardt I think the artifact ID should be fixed now. The other issues can be fixed later, but should not go forgotten.

**Comment by stweil** [comment]
PR #574 supersedes this one, so I maybe PR #571 can be closed. I think that the most important issues addressed here are fixed in PR #574, so that one can be merged. @matthias-ronge, could you please check that, too? @henning-gerhardt, there are still open issues, and as @matthias-ronge already said, those should be tracked and fixed.

**Comment by henning-gerhardt** [comment]
Superseeded by #574.

**Pull request closed without merging**